tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -152,111 +154,322 @@ def assert_alert_rule_trigger_action_migrated(alert_rule_trigger_action, action_
+        )
+        action_alert_rule_trigger_action = ActionAlertRuleTriggerAction.objects.create(
+            action_id=action.id,
+            alert_rule_trigger_action_id=alert_rule_trigger_action.id,
+        )
+        return action, data_condition_group_action, action_alert_rule_trigger_action
+
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.REGION)
+    def create_alert_rule_detector(alert_rule: AlertRule, detector: Detector) -> AlertRuleDetector:
+        return AlertRuleDetector.objects.create(alert_rule=alert_rule, detector=detector)
+
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.REGION)
+    def create_alert_rule_workflow(alert_rule: AlertRule, workflow: Workflow) -> AlertRuleWorkflow:
+        return AlertRuleWorkflow.objects.create(alert_rule=alert_rule, workflow=workflow)
+
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.REGION)
+    def create_action_alert_rule_trigger_action(
+        action: Action, alert_rule_trigger_action: AlertRuleTriggerAction
+    ) -> ActionAlertRuleTriggerAction:
+        return ActionAlertRuleTriggerAction.objects.create(
+            action=action, alert_rule_trigger_action=alert_rule_trigger_action
         )
 
-    def test_create_metric_alert(self):
+
+class DualWriteAlertRuleTest(APITestCase):
+    def setUp(self):
+        self.rpc_user = user_service.get_user(user_id=self.user.id)
+        self.metric_alert = self.create_alert_rule(resolve_threshold=2)
+
+    def test_dual_write_metric_alert(self):
         """
         Test that when we call the helper methods we create all the ACI models correctly for an alert rule
         """
         migrate_alert_rule(self.metric_alert, self.rpc_user)
         assert_alert_rule_migrated(self.metric_alert, self.project.id)
 
-    def test_delete_metric_alert(self):
-        migrate_alert_rule(self.metric_alert, self.rpc_user)
-        alert_rule_workflow = AlertRuleWorkflow.objects.get(alert_rule=self.metric_alert)
-        alert_rule_detector = AlertRuleDetector.objects.get(alert_rule=self.metric_alert)
-        workflow = Workflow.objects.get(id=alert_rule_workflow.workflow.id)
-        detector = Detector.objects.get(id=alert_rule_detector.detector.id)
-        detector_workflow = DetectorWorkflow.objects.get(detector=detector)
-        data_condition_group = detector.workflow_condition_group
-        assert data_condition_group is not None
-        query_subscription = QuerySubscription.objects.get(
-            snuba_query=self.metric_alert.snuba_query.id
-        )
-        data_source = DataSource.objects.get(
-            organization_id=self.metric_alert.organization_id, query_id=query_subscription.id
-        )
-        detector_state = DetectorState.objects.get(detector=detector)
-        data_source_detector = DataSourceDetector.objects.get(data_source=data_source)
 
+class DualDeleteAlertRuleTest(BaseMetricAlertMigrationTest):
+    def setUp(self):
+        self.metric_alert = self.create_alert_rule()
```

Comment:
```suggestion
class DualDeleteAlertRuleTest(BaseMetricAlertMigrationTest):
    def setUp(self):
        super().setUp()
        self.metric_alert = self.create_alert_rule()
```
when overriding a method defined in the base class, it's good practice to invoke the parent classes first to make sure all the prior setup is completed too.